satpy/tests/test_composites.py

```diff
@@ -173,83 +175,153 @@ def setUp(self):
+        [
+            {'name': "true_color", "high_resolution_band": "bad", "neutral_resolution_band": "bad"},
+        ]
+    )
+    def test_bad_colors(self, init_kwargs):
         """Test that only valid band colors can be provided."""
         from satpy.composites import RatioSharpenedRGB
-        self.assertRaises(ValueError, RatioSharpenedRGB, name='true_color', high_resolution_band='bad')
+        with pytest.raises(ValueError):
+            RatioSharpenedRGB(**init_kwargs)
 
-    def test_match_data_arrays(self):
+    @pytest.mark.parametrize(
+        "exp",
+        [
+            satpy.composites.IncompatibleAreas,
+        ]
+    )
+    def test_match_data_arrays(self, exp):
         """Test that all areas have to be the same resolution."""
-        from satpy.composites import IncompatibleAreas, RatioSharpenedRGB
+        from satpy.composites import RatioSharpenedRGB
         comp = RatioSharpenedRGB(name='true_color')
-        self.assertRaises(IncompatibleAreas, comp, (self.ds1, self.ds2, self.ds3), optional_datasets=(self.ds4_big,))
+        with pytest.raises(exp):
+            comp((self.ds1, self.ds2, self.ds3), optional_datasets=(self.ds4_big,))
 
-    def test_more_than_three_datasets(self):
+    @pytest.mark.parametrize(
+        "exp",
+        [
+            ValueError,
+        ]
+    )
+    def test_more_than_three_datasets(self, exp):
         """Test that only 3 datasets can be passed."""
-        from satpy.composites import RatioSharpenedRGB
-        comp = RatioSharpenedRGB(name='true_color')
-        self.assertRaises(ValueError, comp, (self.ds1, self.ds2, self.ds3, self.ds1),
-                          optional_datasets=(self.ds4_big,))
+        from satpy.composites import SelfSharpenedRGB
+        comp = SelfSharpenedRGB(name='true_color', high_resolution_band=None)
+        with pytest.raises(exp):
+            comp((self.ds1, self.ds2, self.ds3))
 
-    def test_basic_no_high_res(self):
-        """Test that three datasets can be passed without optional high res."""
-        from satpy.composites import RatioSharpenedRGB
-        comp = RatioSharpenedRGB(name='true_color')
-        res = comp((self.ds1, self.ds2, self.ds3))
-        self.assertEqual(res.shape, (3, 2, 2))
+    @pytest.mark.parametrize(
+        "exp",
+        [
+            ValueError,
+        ]
+    )
+    def test_self_sharpened_no_high_res(self, exp):
+        """Test for exception when no high_res band is specified."""
+        from satpy.composites import SelfSharpenedRGB
+        comp = SelfSharpenedRGB(name='true_color', high_resolution_band=None)
+        with pytest.raises(exp):
+            comp((self.ds1, self.ds2, self.ds3))
 
-    def test_basic_no_sharpen(self):
-        """Test that color None does no sharpening."""
+    @pytest.mark.parametrize(
+        ("case", "exp"),
+        [
+            ("without optional high res", (3, 2, 2)),
+            ("high res band is None", (3, 2, 2))
+        ]
+    )
```

Comment:
These string cases should be removed and instead pass init_kwargs and maybe a boolean for whether or not optional_datasets should be passed to the compositor.

```
call_kwargs = {}
if pass_opt:
    call_kwargs["optional_datasets = (self.ds4,)
comp = RatioSharpenedRGB(name="true_color", **init_kwargs)
res = comp((self.ds1, self.ds2, self.ds3), **call_kwargs)
```

Reply:
I decide to split them just like orginal ones for better understanding